core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -927,6 +923,18 @@ public Fingerprint(@CheckForNull Run build, @NonNull String fileName, @NonNull b
         return Util.toHexString(md5sum);
     }
 
+    /**
+     * Converts md5 hash string to byte array.
+     *
+     * @since TODO
+     */
+    public static byte[] toByteArray(String md5sum) {
+        byte[] data = new byte[16];
+        for( int i=0; i<md5sum.length(); i+=2 )
+            data[i/2] = (byte)Integer.parseInt(md5sum.substring(i,i+2),16);
```

Comment:
Been a while for such conversions for me, but do you need to watch for any overflows or anything?

Reply:
Hi @afalko, Actually this piece of code I ported from the tests because it has use with other classes too. Ill comment back after having properly investigated it